File: hw5/examples/cpp/mysql_test_1.cpp

#include <iostream>
#include <mysql_connection.h>
#include <cppconn/driver.h>
#include <cppconn/statement.h>
#include <cppconn/resultset.h>
#include <cppconn/exception.h>

#include "config.h"

using namespace std;


int main()
{
  try {

    // connection info
    string hst = config::host;
    string usr = config::user;
    string pwd = config::password;
    string dab = config::database;

    // create connection
    sql::Driver* driver = get_driver_instance();
    sql::Connection* cn = driver->connect(hst, usr, pwd);
    cn->setSchema(dab);

    // create statement
    sql::Statement* st = cn->createStatement();
    string q = "SELECT id, name FROM pet ORDER BY name";
    
    // execute query
    sql::ResultSet* rs = st->executeQuery(q);

    // print result
    while(rs->next())
      cout << rs->getInt("id") << ", " << rs->getString("name") << endl;

    // clean up
    delete rs;
    delete st;
    delete cn;

  } catch(sql::SQLException& e) {
    cout << e.what() << endl;
  }

}

File: hw5/examples/java/MySQLTest1.java

import java.io.FileInputStream;
import java.io.IOException;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.Properties;

public class MySQLTest1 {

  public static void main(String[] args) {

    try {
      // connection info
      Properties prop = new Properties();
      FileInputStream in = new FileInputStream("config.properties");
      prop.load(in);
      in.close();
      
      // connect to datbase
      String hst = prop.getProperty("host");
      String usr = prop.getProperty("user");
      String pwd = prop.getProperty("password");
      String dab = prop.getProperty("database");
      String url = "jdbc:mariadb://" + hst + "/" + dab;
      Connection cn = DriverManager.getConnection(url, usr, pwd);
      
      // create and execute query
      String q = "SELECT id, name FROM pet ORDER BY name";
      Statement st = cn.createStatement();
      ResultSet rs = st.executeQuery(q);
      
      // print results
      while(rs.next()) {
        int id = rs.getInt("id");
        String name = rs.getString("name");
        System.out.println(id + ", " + name);
      }
      
      // release resources
      rs.close();
      st.close();
      cn.close();
    }
    catch(Exception err) {
      err.printStackTrace();
    }
  }
  
}
